setup.cfg

```diff
@@ -35,6 +35,7 @@ typeguard = py.typed
 test =
     pytest
     typing_extensions
+    mypy == 0.790
```

Comment:
New versions of mypy can cause things that previously passed to now fail (this literally just happened to me on another project).  If we don't pin the version and this happens in the future then `master` could suddenly start failing CI without any changes to typeguard.

Reply:
If this happens, it's likely that mypy has fixed a problem that was hidden before. Whenever this happens, it warrants an investigation from our side.

Reply:
Agreed.  The question is whether you'd prefer to require this investigation to happen immediately because CI is failing on `master`, or within the context of a PR when someone updates the pinned version here.  I don't have an opinion either way; we can do whatever you'd like.  Do you want to remove the version pin?

Reply:
Yes, please.

Reply:
Done